tests/unit/src/test/scala/tests/DocumentSymbolSlowSuite.scala

```diff
@@ -0,0 +1,66 @@
+package tests
+
+object DocumentSymbolSlowSuite extends BaseSlowSuite("documentSymbol") {
+
+  testAsync("parse-error") {
+    for {
+      // start with code that does not parse (notice the first char in Main.scala)
+      _ <- server.initialize(
+        """|
+           |/metals.json
+           |{
+           |  "a": { }
+           |}
+           |/a/src/main/scala/a/Main.scala
+           |} // <- parse error
+           |object Outer {
+           |  class Inner
+           |}
+           |""".stripMargin
+      )
+      _ <- server.didOpen("a/src/main/scala/a/Main.scala")
+      // check that no document symbols have been found for the unparseable code
+      _ = assertNoDiff(
+        server.documentSymbols("a/src/main/scala/a/Main.scala"),
+        """|} // <- parse error
+           |object Outer {
+           |  class Inner
+           |}""".stripMargin
+      )
+      // fix the code to make it parse
+      _ <- server.didChange("a/src/main/scala/a/Main.scala") { text =>
+        text.replaceFirst("}", "")
+      }
+      // check that all document symbols have been found
+      _ = assertNoDiff(
+        server.documentSymbols("a/src/main/scala/a/Main.scala"),
+        """| // <- parse error
+           |/*Outer:3*/object Outer {
+           |  /*Inner:2*/class Inner
+           |}""".stripMargin
+      )
+      // make the code unparseable again
+      _ <- server.didChange("a/src/main/scala/a/Main.scala")(text => text)
```

Comment:
why is it unparseable again? I expected text to have the `}` removed

Reply:
I think it's a "bug" with `didChange`. It seems to always return the original buffer

Reply:
That makes sense, would be good to fix `didChange` so `text` has the buffer value

Reply:
I'll try :)

Reply:
Done 👍